### PR TITLE
Hotfix for multiple events with recursive merge applicable objects

### PR DESCRIPTION
**Pull request by george-clements**
Hotfix for multiple events with objects requiring disabled recursive merge
Updated tests
Updated logic to protect against multiple firings

### CODE REVIEW

File: template.tpl [hunk 1]
```diff
@@ -108,37 +109,44 @@ if (!alreadyFired) {
...
+        dataLayer.forEach(function(element) {
             //Clear the event ID so that the events are treated as new events
             element["gtm.uniqueEventId"] = undefined;
             //Set pushType to artificial to allow for easier debugging
             element.pushType = "artificial";
             log('Repeat DataLayer Events Tag - Repushing event: ' + element.event);
             //Repush the event
-            callLater(()=>dataLayerPush(element));
+            callLater(()=>
+                      recursive_merge(element) && dataLayerPush(recursive_merge(element))
```
Comment (benedikt-buchert) [inline template.tpl:138]: You could use a variable here:

```
const null_push = recursive_merge(element);
null_push &&  dataLayerPush(null_push);
```
It's more DRY.

File: template.tpl [hunk 2]
```diff
@@ -332,14 +340,12 @@ scenarios:
     ];
 
     mock('copyFromWindow', key => {
```
Comment (benedikt-buchert) [inline template.tpl:342]: This is now a duplicate of what you are doing bellow.

Comment (benedikt-buchert) [reply template.tpl:342]: Line 390

**Review by benedikt-buchert — CHANGES_REQUESTED** [review]
See comments in the request.

**Comment by benedikt-buchert** [comment]
Everything else looks good and seems to be working.

File: template.tpl (hunk 2 above)
Comment (george-clements) [reply template.tpl:342]: Yes, but these are two separate tests

**Pull request merged**